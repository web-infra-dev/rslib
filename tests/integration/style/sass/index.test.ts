import { join } from 'node:path';
import { buildAndGetResults } from 'test-helper';
import { expect, test } from 'vitest';

test('should extract css with pluginSass in bundle', async () => {
  const fixturePath = join(__dirname, 'bundle');
  const { contents } = await buildAndGetResults({ fixturePath, type: 'css' });

  const esmFileNames = Object.keys(contents.esm);
  const esmFileContents = Object.values(contents.esm);
  expect(esmFileNames).toMatchInlineSnapshot(`
    [
      "<ROOT>/tests/integration/style/sass/bundle/dist/esm/index.css",
    ]
  `);

  expect(esmFileContents).toMatchSnapshot();

  const cjsFileNames = Object.keys(contents.cjs);
  const cjsFileContents = Object.values(contents.cjs);
  expect(cjsFileNames).toMatchInlineSnapshot(`
    [
      "<ROOT>/tests/integration/style/sass/bundle/dist/cjs/index.css",
    ]
  `);
  expect(cjsFileContents).toMatchSnapshot();
});

test('should extract css with pluginSass in bundle-false', async () => {
  const fixturePath = join(__dirname, 'bundle-false');
  const { contents } = await buildAndGetResults({ fixturePath, type: 'css' });

  const esmFileNames = Object.keys(contents.esm);
  const esmFileContents = Object.values(contents.esm);
  expect(esmFileNames).toMatchInlineSnapshot(`
    [
      "<ROOT>/tests/integration/style/sass/bundle-false/dist/esm/foundation/_code.css",
      "<ROOT>/tests/integration/style/sass/bundle-false/dist/esm/foundation/_lists.css",
      "<ROOT>/tests/integration/style/sass/bundle-false/dist/esm/foundation/index.css",
      "<ROOT>/tests/integration/style/sass/bundle-false/dist/esm/index.css",
    ]
  `);
  expect(esmFileContents).toMatchSnapshot();

  const cjsFileNames = Object.keys(contents.cjs);
  const cjsFileContents = Object.values(contents.cjs);
  expect(cjsFileNames).toMatchInlineSnapshot(`
    [
      "<ROOT>/tests/integration/style/sass/bundle-false/dist/cjs/foundation/_code.css",
      "<ROOT>/tests/integration/style/sass/bundle-false/dist/cjs/foundation/_lists.css",
      "<ROOT>/tests/integration/style/sass/bundle-false/dist/cjs/foundation/index.css",
      "<ROOT>/tests/integration/style/sass/bundle-false/dist/cjs/index.css",
    ]
  `);
  expect(cjsFileContents).toMatchSnapshot();
});
