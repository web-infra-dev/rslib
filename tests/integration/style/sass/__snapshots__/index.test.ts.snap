// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should extract css with pluginSass in bundle 2`] = `
[
  ".lib1 {
  font-size: 18px;
}

code {
  padding: .25em;
  line-height: 0;
}

ul, ol {
  text-align: left;
}

ul ul, ul ol, ol ul, ol ol {
  padding-bottom: 0;
  padding-left: 0;
}

body, .url-variable {
  background: url(./static/svg/logo.svg);
}

.alert {
  border: 1px solid #c6538ce0;
}

",
]
`;

exports[`should extract css with pluginSass in bundle 4`] = `
[
  ".lib1 {
  font-size: 18px;
}

code {
  padding: .25em;
  line-height: 0;
}

ul, ol {
  text-align: left;
}

ul ul, ul ol, ol ul, ol ol {
  padding-bottom: 0;
  padding-left: 0;
}

body, .url-variable {
  background: url(./static/svg/logo.svg);
}

.alert {
  border: 1px solid #c6538ce0;
}

",
]
`;

exports[`should extract css with pluginSass in bundle-false 2`] = `
[
  "code {
  padding: .25em;
  line-height: 0;
}

",
  "ul, ol {
  text-align: left;
}

ul ul, ul ol, ol ul, ol ol {
  padding-bottom: 0;
  padding-left: 0;
}

",
  "body {
  background: url(../static/svg/logo.svg);
}

",
  "@import "~lib1/index.css";

code {
  padding: .25em;
  line-height: 0;
}

ul, ol {
  text-align: left;
}

ul ul, ul ol, ol ul, ol ol {
  padding-bottom: 0;
  padding-left: 0;
}

body, .url-variable {
  background: url(./static/svg/logo.svg);
}

.alert {
  border: 1px solid #c6538ce0;
}

",
]
`;

exports[`should extract css with pluginSass in bundle-false 4`] = `
[
  "code {
  padding: .25em;
  line-height: 0;
}

",
  "ul, ol {
  text-align: left;
}

ul ul, ul ol, ol ul, ol ol {
  padding-bottom: 0;
  padding-left: 0;
}

",
  "body {
  background: url(../static/svg/logo.svg);
}

",
  "@import "~lib1/index.css";

code {
  padding: .25em;
  line-height: 0;
}

ul, ol {
  text-align: left;
}

ul ul, ul ol, ol ul, ol ol {
  padding-bottom: 0;
  padding-left: 0;
}

body, .url-variable {
  background: url(./static/svg/logo.svg);
}

.alert {
  border: 1px solid #c6538ce0;
}

",
]
`;
